shapefile.py

```diff
@@ -553,9 +545,9 @@ def __dbfHeader(self):
             else:
                 idx = len(fieldDesc[name]) - 1
             fieldDesc[name] = fieldDesc[name][:idx]
-            fieldDesc[name] = u(fieldDesc[name])
+            fieldDesc[name] = u(fieldDesc[name], self.encoding)
             fieldDesc[name] = fieldDesc[name].lstrip()
-            fieldDesc[1] = u(fieldDesc[1])
+            fieldDesc[1] = u(fieldDesc[1], self.encoding)
             self.fields.append(fieldDesc)

```

Comment:
If I have read the DBF specification correctly ([link](http://www.oocities.org/geoff_wass/dBASE/GaryWhite/dBASE/FAQ/qformt.htm#D)), field names can only contain and be stored as ASCII characters. If so, I guess these two lines should use "ascii" as encoding instead of self.encoding?